appserver/extras/arquillian-containers/payara-common/src/main/java/fish/payara/arquillian/container/payara/CommonPayaraManager.java

```diff
@@ -0,0 +1,165 @@
+            final FormDataMultiPart form = new FormDataMultiPart();
+            form.bodyPart(new StreamDataBodyPart("id", deployment, archiveName));
+
+            deploymentName = createDeploymentName(archiveName);
+            addDeployFormFields(deploymentName, form);
+
+            // Do Deploy the application on the remote Payara
+            HTTPContext httpContext = glassFishClient.doDeploy(deploymentName, form);
+            protocolMetaData.addContext(httpContext);
+        } catch (PayaraClientException e) {
+            throw new DeploymentException("Could not deploy " + archiveName, e);
+        }
+        
+        return protocolMetaData;
+    }
+
+    public void undeploy(Archive<?> archive) throws DeploymentException {
+
+        if (archive == null) {
+            throw new IllegalArgumentException("archive must not be null");
+        }
+
+        deploymentName = createDeploymentName(archive.getName());
+        
+        try {
+            // Build up the POST form to send to Glassfish
+            FormDataMultiPart form = new FormDataMultiPart();
+            form.field("target", configuration.getTarget(), TEXT_PLAIN_TYPE);
+            form.field("operation", DELETE_OPERATION, TEXT_PLAIN_TYPE);
+            
+            glassFishClient.doUndeploy(deploymentName, form);
+        } catch (PayaraClientException e) {
+            throw new DeploymentException("Could not undeploy " + archive.getName(), e);
+        }
+    }
+
+    public boolean isDASRunning() {
+        return glassFishClient.isDASRunning();
+    }
+
+    private String createDeploymentName(String archiveName) {
+        String correctedName = archiveName;
+        if (correctedName.startsWith("/")) {
+            correctedName = correctedName.substring(1);
+        }
+        
+        if (correctedName.indexOf(".") != -1) {
+            correctedName = correctedName.substring(0, correctedName.lastIndexOf("."));
+        }
+        
+        return correctedName;
+    }
+
+    private void addDeployFormFields(String name, FormDataMultiPart deployform) {
+
+        // Add the name field, the name is the archive filename without extension
+        deployform.field("name", name, TEXT_PLAIN_TYPE);
+
+        // Add the target field (the default is "server" - Admin Server)
+        deployform.field("target", this.configuration.getTarget(), TEXT_PLAIN_TYPE);
+
+        // Add the libraries field (optional)
+        if (configuration.getLibraries() != null) {
+            deployform.field("libraries", configuration.getLibraries(), TEXT_PLAIN_TYPE);
+        }
+
+        // Add the properties field (optional)
+        if (configuration.getProperties() != null) {
+            deployform.field("properties", this.configuration.getProperties(), TEXT_PLAIN_TYPE);
+        }
+
+        // Add the type field (optional, the only valid value is "osgi", other values are ommited)
+        if (this.configuration.getType() != null && "osgi".equals(configuration.getType())) {

```

Comment:
I love nothing better, a bunch of those were already done, but I clearly missed this one.